### ISSUE TITLE
feature: Need Documentation for KCP Installation

### ISSUE THREAD

**Issue by ChocolatePadmanaban** [kcp-dev/kcp#3100]
### Feature Description

 Need Documentation for KCP Installation

### Proposed Solution

ReadMe to have step kcp installation

### Want to contribute?

- [X] I would like to work on this issue.

**Comment by embik** [comment]
Hey @ChocolatePadmanaban, thanks for opening this ticket. I see that you have checked the checkbox for contributing this yourself, I'm happy to support you if you're up for it.

**Comment by embik** [comment]
Hi @ChocolatePadmanaban, following up on this. #3113 added a new "Setup" part of our documentation, I've briefly mentioned the Helm chart there. If you _are_ interested in this, feel free to extend those pages.

**Comment by rosieks** [comment]
It would be really great if you could improve installation guide. Here are some findings that I notice:
* **Quickstart**: Address how to run it locally on Windows. I noticed there is no kcp for windows so probably there should be information how to run it on minikube or something like that.
* **Installation with Helm**: Provide a command that can be copy-paste and just works. Right now it refers to `my-values.yaml` that is not provided. It would be really great to have two vaules file. Minimal and just sample.

**Comment by embik** [comment]
> Installation with Helm: Provide a command that can be copy-paste and just works. Right now it refers to my-values.yaml that is not provided. It would be really great to have two vaules file. Minimal and just sample.

The main problem is we cannot provide a file that just works because the external hostname needs to be a domain you control. So people will _have_ to create/modify their own `values.yaml` because they have to deal with DNS.

**Comment by rosieks** [comment]
Can it be solved by adding something to hosts file?

**Comment by ntnn** [comment]
Absolutely, if you look at the tilt config it expects `kcp.dev.local` and `idp.dev.local` for the resulting kubeconfig:

https://github.com/kcp-dev/kcp/blob/1933949229d8c9b85a0189651c779836b065fcc0/contrib/tilt/kcp-values.yaml#L1

**Comment by embik** [comment]
There _is_ also https://github.com/kcp-dev/helm-charts/tree/main/charts/kcp#install-to-kind-cluster-for-development in the Helm chart README for setting up kcp locally in kind. It is not suitable for installing kcp on a real Kubernetes cluster, but was the kind of thing you had in mind @rosieks?

**Comment by rosieks** [comment]
Yes, exactly the same although for local kubernetes (eg. minikube). Extra points if you can run it on windows nativly (so any script is in batch or powershell)

**Comment by embik** [comment]
kind is local Kubernetes. I don't believe any of the maintainers are on Windows but we would welcome contributions that help getting started with kcp on Windows.

We should extend the quickstart guide with this for sure, so will take a look at doing that.
